# 16S_Analysis_Workflow
This workflow has step by step instructions and scripts to 16S microbiome analysis.

## Workflow Overview
<<<<<<< HEAD
### Key steps, Input files, Required tools/packages
#### Input files:
1. paired fastq files
2. Metadata: This should be verified using Keemei and should be in tsv format

#### Tools to be installed:
1. Qiime2
2. Phyloseq. It is an R package

#### Key steps:
1. Generating a manifest file and importing data into qiime2
2. Qaulity control to remove poor quality reads, adapters, chimeric reads etc
3. Generating a feature table (ASV file).
4. Phylogeny and diversity analysis
5. Taxonomic analysis

### Downstream Analysis using phyloseq
1. Data exploration and quality check
=======
Created by Monica M
>>>>>>> 5b67a24a8768504f8a7a52fad58b16ff4e9748fd
